src/Image.tsx

```diff
@@ -1,59 +1,49 @@
-import React from "react";
-import FastImage from "react-native-fast-image";
-import { ImageStyle, ImageURISource } from "react-native";
+import React, { FC, useState } from "react";
+import FastImage, { FastImageProps } from "react-native-fast-image";
+import { ImageStyle } from "react-native";
 
-export interface ImageProps {
+export interface ImageProps extends FastImageProps {
   width?: number;
-  style: ImageStyle | ImageStyle[];
-  source: ImageURISource;
+  style?: ImageStyle | ImageStyle[];
   onLoad?: (args: any) => any;
 }
 
 export interface ImageState {
   height: number;
 }
 
-class Wrap extends React.PureComponent<ImageProps, ImageState> {
-  state = {
-    height: 150,
-  };
-
-  handleLoad = (tempWidth: number, tempHeight: number) => {
-    const { width } = this.props;
+const Wrap: FC<ImageProps> = ({ style, width, onLoad, ...rest }) => {
+  const [height, setHeight] = useState(150);
 
+  const handleLoad = (tempWidth: number, tempHeight: number) => {
     if (width) {
       const ratio = tempHeight / tempWidth;
       const height = width * ratio;
-      this.setState({
-        height,
-      });
+
+      setHeight(height);
     }
   };
 
-  render() {
-    const { width, source, onLoad = () => {}, style = {} } = this.props;
-    const { height } = this.state;
-    return (
-      <FastImage
-        style={[
-          {
-            width,
-            height,
-          },
-          style,
-        ]}
-        source={{
-          uri: source.uri,
-          priority: FastImage.priority.high,
-        }}
-        onLoad={(e) => {
-          this.handleLoad(e.nativeEvent.width, e.nativeEvent.height);
+  return (
+    <FastImage
+      style={[
+        {
+          width,
+          height,
+        },
+        style,
+      ]}
+      onLoad={(e: any) => {
+        handleLoad(e.nativeEvent.width, e.nativeEvent.height);
+
+        if (!!onLoad) {
           onLoad(e);
-        }}
-        resizeMode={FastImage.resizeMode.cover}
-      />
-    );
-  }
-}
+        }
+      }}
+      resizeMode={FastImage.resizeMode.cover}
+      {...rest}
```

Comment:
I don't like having these spread props, makes it impossible to know what props you're using. Do you think it's better?

Reply:
I assume that it can be better because right now we extend from `FastImageProps` and this `FastImageProps` may change, so instead of mapping props manually it is way more easier to spread them and we will get the latest props from `react-native-fast-image`. 

But I can also map them manually if you think it is better 👍

Reply:
Good point, and as long as the types are clear when you use the component, thats fine!

Reply:
I assume that it can be better because right now we extend from FastImageProps and this FastImageProps may change, so instead of mapping props manually it is way more easier to spread them and we will get the latest props from react-native-fast-image.

But I can also map them manually if you think it is better 👍